### PR TITLE
Skip region check if we use shard pinning

### PR DESCRIPTION
**Pull request by andreadecorte**
This is consistent with the behaviour on backend side, which will also check the authorization of using it

### CODE REVIEW

File: pkg/properties/properties.go [hunk 1]
```diff
@@ -32,3 +32,5 @@ const FakeCluster = "fake_cluster"
 
 // nolint:gosec // Linter thinks there are hardcoded credentials here...
 const UseLocalCredentials = "use_local_credentials"
+
+const ProvisionShardId = "provision_shard_id"
```
Comment (lucasponce) [inline pkg/properties/properties.go:36]: This property is for developer scenarios, right?
I mean, do we expect that users require to add the provision_shard_id ?

Comment (andreadecorte) [reply pkg/properties/properties.go:36]: it is optional (providing properties is an hidden flag). This PR doesn't change anything if we are not providing something like
--properties provision_shard_id:xxx in input

File: cmd/create/cluster/cmd.go [hunk 2]
```diff
@@ -1237,7 +1245,7 @@ func run(cmd *cobra.Command, _ []string) {
 		r.Reporter.Errorf("Expected a valid AWS region")
 		os.Exit(1)
 	} else {
-		if isHostedCP && !interactive.Enabled() {
+		if isHostedCP && !interactive.Enabled() && !shardPinningEnabled {
```
Comment (tiwillia) [inline cmd/create/cluster/cmd.go:1248]: I think this might be incorrect. Why is this region check being done only when shad pinning is _not_ enabled?

**Pull request merged**